### PR TITLE
Display a tooltip when you hover over a location

### PR DESCRIPTION
**Pull request by andybalaam**
Builds on https://github.com/matrix-org/matrix-react-sdk/pull/7465.

Show "Expand map" tooltip on a map.

image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Display a tooltip when you hover over a location ([\#7472](https://github.com/matrix-org/matrix-react-sdk/pull/7472)). Contributed by @andybalaam.

Preview: https://61d8575b1b79672ab17ca36e--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/messages/MLocationBody.tsx [hunk 1]
```diff
@@ -93,16 +95,19 @@ export default class MLocationBody extends React.Component<IBodyProps, IState> {
             bodyId={this.getBodyId()}
             markerId={this.getMarkerId()}
             error={this.state.error}
+            tooltip={_t("Expand map")}
             onClick={this.onClick}
         />;
     }
 }
 
+
```
Comment (t3chguy) [inline src/components/views/messages/MLocationBody.tsx]: redundant two blank lines?

File: src/components/views/messages/MLocationBody.tsx [hunk 2]
```diff
@@ -93,16 +95,19 @@ export default class MLocationBody extends React.Component<IBodyProps, IState> {
...
 interface ILocationBodyContentProps {
     mxEvent: MatrixEvent;
     bodyId: string;
     markerId: string;
     error: Error;
+    tooltip?: String;
```
Comment (t3chguy) [inline src/components/views/messages/MLocationBody.tsx:109]:
```suggestion
    tooltip?: string;
```

Comment (t3chguy) [reply src/components/views/messages/MLocationBody.tsx:109]: Primitive vs boxed String instance

File: src/components/views/messages/MLocationBody.tsx [hunk 3]
```diff
@@ -115,11 +120,25 @@ export function LocationBodyContent(props: ILocationBodyContentProps) {
...
-            className="mx_MLocationBody_map"
-        />
+        {
+            props.tooltip
+                ? <TooltipTarget
+                    label={props.tooltip}
+                    alignment={Alignment.InnerBottom}
+                    maxParentWidth={450}
+                >
+                        <div
```
Comment (t3chguy) [inline src/components/views/messages/MLocationBody.tsx]:
```suggestion
                    <div
```

File: src/components/views/messages/MLocationBody.tsx [hunk 4]
```diff
@@ -115,11 +120,25 @@ export function LocationBodyContent(props: ILocationBodyContentProps) {
...
+                        id={props.bodyId}
+                        onClick={props.onClick}
+                        className="mx_MLocationBody_map"
+                    />
+                </TooltipTarget>
+                : <div
+                    id={props.bodyId}
+                    onClick={props.onClick}
+                    className="mx_MLocationBody_map"
+                />
```
Comment (t3chguy) [inline src/components/views/messages/MLocationBody.tsx:139]: kinda feels like this ought to be a const set earlier and re-used in both branches of this ternary

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/components/views/messages/MLocationBody.tsx (hunk 2 above)
Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx]: Sorry, was writing Rust this morning :-)

File: src/components/views/messages/MLocationBody.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx]: Fixed in 04aebeb0e0ed975a45f49191f8830fca949bda45

File: src/components/views/messages/MLocationBody.tsx (hunk 3 above)
Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx]: Fixed in 04aebeb0e0ed975a45f49191f8830fca949bda45

File: src/components/views/messages/MLocationBody.tsx (hunk 4 above)
Comment (andybalaam) [reply src/components/views/messages/MLocationBody.tsx]: Fixed in 7d6e07c392

**Review by t3chguy — APPROVED** [review]
Looks sane, the tooltip stuff is a mess in general though sadly :'(

**Comment by t3chguy** [comment]
Blocked on https://github.com/matrix-org/matrix-react-sdk/pull/7465

**Pull request merged**